# COGS 108 - Data Checkpoint

## Authors

Instructions: REPLACE the contents of this cell with your team list and their contributions. Note that this will change over the course of the checkpoints

This is a modified [CRediT taxonomy of contributions](https://credit.niso.org). For each group member please list how they contributed to this project using these terms:
> Analysis, Background research, Conceptualization, Data curation, Experimental investigation, Methodology, Project administration, Software, Visualization, Writing – original draft, Writing – review & editing

Example team list and credits:
- Alice Anderson: Conceptualization, Data curation, Methodology, Writing - original draft
- Bob Barker:  Analysis, Software, Visualization
- Charlie Chang: Project administration, Software, Writing - review & editing
- Dani Delgado: Analysis, Background research, Visualization, Writing - original draft

## Research Question

Among full-time undergraduate students enrolled in on-campus, in-person courses, is country GDP per capita (PPP) associated with **(1) students’ self-reported frequency of ChatGPT use for schoolwork (Rarely–Extensively)** and **(2) whether students use ChatGPT relatively more for schoolwork learning support (e.g., explanations, summaries, study help) versus schoolwork assignment production (e.g., writing, coding, generating deliverables)**?

## Background and Prior Work

AI has the potential to improve a wide range of fields and these industries may vary depending on the country. For example, in Kenya, AI is utilized to analyze the soil and crop conditions in agriculture.<a name="cite_ref-1"></a>[<sup>1</sup>](#cite_note-1) On the other hand, the healthcare industry in South Africa utilizes AI to better understand and meet patient needs.<a name="cite_ref-1"></a>[<sup>1</sup>](#cite_note-1) AI powered tools have substantially reduced human errors in such fields which require precise calculations, leading to a rapid development of new technology, thus enhancing a country’s economic growth. Vice versa, the economic state of a country has the potential to shape how AI is used.

While AI has been a tool to improve the economic state of a country, it also has improved our daily lives by conveniently summarizing information at the tips of our fingers. With an increase in AI investment contributing significantly to the GDP growth since 2014<a name="cite_ref-2"></a>[<sup>2</sup>](#cite_note-2), a majority of students from various fields, surveyed between late 2023 and early 2024, believe that using AI improves learning and assignment execution in general<a name="cite_ref-3"></a>[<sup>3</sup>](#cite_note-3). However, a portion of the surveyed students expressed concerns over AI’s reliability and ethicalness within the academic setting.

Despite its benefit for the overall global development, AI has the potential to backfire. With an increase in AI usage comes an increase in concerns over how AI is utilized as a tool. In prior research and analysis, results reflected the trade off of using generative AI, a declination in critical thinking. In the study conducted by Kosmyna et al<a name="cite_ref-4"></a>[<sup>4</sup>](#cite_note-4), essays produced by participants in the LLM group reflected weak variability compared to the participants in the Brain-only group and the Search Engine group.

With the variables of economic utilizations of AI and AI’s effects on human cognition, does a country’s development determine AI usage, especially in students? Specifically, does a country’s development relate to student usage of AI, the ways in which it is used as well as how much it is used?


1. <a name="cite_note-1"></a> [^](#cite_ref-1) Folorunso, A., Olanipekun, K., Adewumi, T., & Samuel, B. (2024). *A policy framework on AI usage in developing countries and its impact. Global Journal of Engineering and Technology Advances, 21(01), 154–166.* https://www.researchgate.net/profile/Adebola-Folorunso/publication/385559435_A_policy_framework_on_AI_usage_in_developing_countries_and_its_impact/links/672ac69d5852dd723caaf782/A-policy-framework-on-AI-usage-in-developing-countries-and-its-impact.pdf 
2. <a name="cite_note-2"></a> [^](#cite_ref-2) Rubinton, H., & Patro, B. A. (2026, January 12). *Tracking AI’s contribution to GDP growth. Federal Reserve Bank of St. Louis, On the Economy.* https://www.stlouisfed.org/on-the-economy/2026/jan/tracking-ai-contribution-gdp-growth
3. <a name="cite_note-3"></a> [^](#cite_ref-3) Ravšelj, D., Keržič, D., Tomaževič, N., Umek, L., Brezovar, N., & Aristovnik, A. (2025). *Higher education students’ perceptions of ChatGPT: A global study of early reactions.* PLOS ONE, 20(2), e0315011. https://doi.org/10.1371/journal.pone.0315011
4. <a name="cite_note-4"></a> [^](#cite_ref-4) Kosmyna, N., Hauptmann, E., Yuan, Y. T., Situ, J., Liao, X.-H., Beresnitzky, A. V., Braunstein, I., & Maes, P. (2025). Your brain on ChatGPT: Accumulation of cognitive debt when using an AI assistant for essay writing task (arXiv:2506.08872v1) [Preprint]. arXiv. https://arxiv.org/abs/2506.08872v1

## Hypothesis


Hypothesis: Country economic development will predict a dual pattern in ChatGPT usage: higher-income countries will show both greater usage frequency and higher allocation to learning support (tutoring/explanations/summaries), while lower-income countries will show lower frequency but proportionally more assignment production (drafting/writing/coding/generating deliverables).



Explanation:We predict that higher-education students in more economically developed countries will report using ChatGPT more often for schoolwork. We also predict that, among users, students in less developed countries will use ChatGPT relatively more for assignment production (drafting/writing/coding), while students in more developed countries will use it relatively more for learning support (tutoring/explanations/summaries). We expect this pattern because access and adoption tend to be higher in wealthier countries, while higher academic pressure and fewer support resources may push students in lower-income settings toward using ChatGPT to generate deliverables, and stricter academic integrity norms in some higher-income settings may discourage production-style use.


## Data

### Data overview

Instructions: REPLACE the contents of this cell with descriptions of your actual datasets.

For each dataset include the following information
- Dataset #1
  - Dataset Name:
  - Link to the dataset:
  - Number of observations:
  - Number of variables:
  - Description of the variables most relevant to this project
  - Descriptions of any shortcomings this dataset has with repsect to the project
- Dataset #2 (if you have more than one!)
  - same as above
- etc

Each dataset deserves either a set of bullet points as above or a few sentences if you prefer that method.

If you plan to use multiple datasets, add a few sentences about how you plan to combine these datasets.

In [ ]:
# Run this code every time when you're actively developing modules in .py files.  It's not needed if you aren't making modules
#
## this code is necessary for making sure that any modules we load are updated here 
## when their source code .py files are modified

%load_ext autoreload
%autoreload 2

In [ ]:
# Setup code -- this only needs to be run once after cloning the repo!
# this code downloads the data from its source to the `data/00-raw/` directory
# if the data hasn't updated you don't need to do this again!

# if you don't already have these packages (you should!) uncomment this line
# %pip install requests tqdm

import sys
sys.path.append('./modules') # this tells python where to look for modules to import

import get_data # this is where we get the function we need to download data

# replace the urls and filenames in this list with your actual datafiles
# yes you can use Google drive share links or whatever
# format is a list of dictionaries; 
# each dict has keys of 
#   'url' where the resource is located
#   'filename' for the local filename where it will be stored 
datafiles = [
    { 'url': 'https://raw.githubusercontent.com/fivethirtyeight/data/refs/heads/master/airline-safety/airline-safety.csv', 'filename':'airline-safety.csv'},
    { 'url': 'https://raw.githubusercontent.com/fivethirtyeight/data/refs/heads/master/bad-drivers/bad-drivers.csv', 'filename':'bad-drivers.csv'}
]

get_data.get_raw(datafiles,destination_directory='data/00-raw/')

### Dataset #1 

Instructions: 
1. Change the header from Dataset #1 to something more descriptive of the dataset
2. Write a few paragraphs about this dataset. Make sure to cover
   1. Describe the important metrics, what units they are in, and giv some sense of what they mean.  For example "Fasting blood glucose in units of mg glucose per deciliter of blood.  Normal values for healthy individuals range from 70 to 100 mg/dL.  Values 100-125 are prediabetic and values >125mg/dL indicate diabetes. Values <70 indicate hypoglycemia. Fasting idicates the patient hasn't eaten in the last 8 hours.  If blood glucose is >250 or <50 at any time (regardless of the time of last meal) the patient's life may be in immediate danger"
   2. If there are any major concerns with the dataset, describe them. For example "Dataset is composed of people who are serious enough about eating healthy that they voluntarily downloaded an app dedicated to tracking their eating patterns. This sample is likely biased because of that self-selection. These people own smartphones and may be healthier and may have more disposable income than the average person.  Those who voluntarily log conscientiously and for long amounts of time are also likely even more interested in health than those who download the app and only log a bit before getting tired of it"
3. Use the cell below to 
    1. load the dataset 
    2. make the dataset tidy or demonstrate that it was already tidy
    3. demonstrate the size of the dataset
    4. find out how much data is missing, where its missing, and if its missing at random or seems to have any systematic relationships in its missingness
    5. find and flag any outliers or suspicious entries
    6. clean the data or demonstrate that it was already clean.  You may choose how to deal with missingness (dropna of fillna... how='any' or 'all') and you should justify your choice in some way
    7. You will load raw data from `data/00-raw/`, you will (optionally) write intermediate stages of your work to `data/01-interim` and you will write the final fully wrangled version of your data to `data/02-processed`
4. Optionally you can also show some summary statistics for variables that you think are important to the project
5. Feel free to add more cells here if that's helpful for you


In [106]:
## YOUR CODE TO LOAD/CLEAN/TIDY/WRANGLE THE DATA GOES HERE
import pandas as pd
import numpy as np
import os

A. The data set has been loaded into a pandas DataFrame called `df`. The first few rows of the DataFrame have been displayed using `df.head()`.

In [105]:
# paths
raw_path = 'data/00-raw/early_perceptions_chatgpt.xlsx'
interim_path = 'data/01-interim/early_perceptions_chatgpt_interim.xlsx'
processed_path = 'data/02-processed/early_perceptions_chatgpt_processed.xlsx'

df = pd.read_excel(raw_path)
df.head()

Q1   Q2     Q3           Q4  \
0  Afghanistan  2.0   27.0  Afghanistan   
1  Afghanistan  4.0  100.0  Afghanistan   
2  Afghanistan  1.0   21.0      Albania   
3  Afghanistan  1.0   18.0      Ecuador   
4  Afghanistan  2.0   26.0        Spain   

                                                  Q5   Q6   Q7   Q8   Q9  Q10  \
0                                     القدس المفتوحة  1.0  1.0  1.0  1.0  1.0   
1                                                456  2.0  2.0  3.0  2.0  4.0   
2                                               منيا  1.0  1.0  1.0  1.0  1.0   
3  Universidad Indoamerica., Facultad Arquitectur...  2.0  1.0  1.0  1.0  4.0   
4                                De vigo universidad  1.0  2.0  2.0  1.0  2.0   

   ...  Q40g  Q40h  Q40i  Q40j  Q40k  Q40l  Q40m  Q40n  Q40o  source  
0  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN      AR  
1  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN      EN  
2  ...   3.0   2.0   3.0   4.0   3.0   2.0   2.0   1.0   1.0      EN  
3  ...   3.0   3.0   3.0   3.0   3.0   3.0   3.0   3.0   3.0      ES  
4  ...   4.0   3.0   4.0   3.0   3.0   3.0   4.0   3.0   4.0      ES  

[5 rows x 174 columns]

B. The data set is tidy. There is one entry in each row and column space.

C. The shape is 23218 rows and 174 columns.

In [94]:
print('Shape:', df.shape)
print('Column names:', df.columns.tolist())
print(df.info())

Shape: (23218, 174)
Column names: ['Q1', 'Q2', 'Q3', 'Q4', 'Q5', 'Q6', 'Q7', 'Q8', 'Q9', 'Q10', 'Q11', 'Q12', 'Q13', 'Q14', 'Q15', 'Q16', 'Q17', 'Q18a', 'Q18b', 'Q18c', 'Q18d', 'Q18e', 'Q18f', 'Q18g', 'Q18h', 'Q18i', 'Q18j', 'Q18k', 'Q18l', 'Q19a', 'Q19b', 'Q19c', 'Q19d', 'Q19e', 'Q19f', 'Q19g', 'Q19h', 'Q19i', 'Q19j', 'Q20', 'Q21a', 'Q21b', 'Q21c', 'Q21d', 'Q22a', 'Q22b', 'Q22c', 'Q22d', 'Q22e', 'Q22f', 'Q22g', 'Q22h', 'Q22i', 'Q22j', 'Q23a', 'Q23b', 'Q23c', 'Q23d', 'Q24a', 'Q24b', 'Q24c', 'Q24d', 'Q24e', 'Q24f', 'Q24g', 'Q25a', 'Q25b', 'Q25c', 'Q25d', 'Q26a', 'Q26b', 'Q26c', 'Q26d', 'Q26e', 'Q26f', 'Q26g', 'Q26h', 'Q26i', 'Q26j', 'Q27a', 'Q27b', 'Q27c', 'Q27d', 'Q27e', 'Q27f', 'Q27g', 'Q27h', 'Q27i', 'Q27j', 'Q28a', 'Q28b', 'Q28c', 'Q28d', 'Q28e', 'Q28f', 'Q28g', 'Q28h', 'Q28i', 'Q29a', 'Q29b', 'Q29c', 'Q29d', 'Q29e', 'Q29f', 'Q29g', 'Q29h', 'Q29i', 'Q30a', 'Q30b', 'Q30c', 'Q30d', 'Q30e', 'Q30f', 'Q30g', 'Q30h', 'Q30i', 'Q30j', 'Q30k', 'Q30l', 'Q31a', 'Q31b', 'Q31c', 'Q31d', 'Q32a', 

In [95]:
rename_dict = {

    # SECTION 1 – DEMOGRAPHICS
    'Q1': 'citizenship',
    'Q2': 'gender',
    'Q3': 'age',
    'Q4': 'country_study',
    'Q5': 'institution',
    'Q6': 'public_institution',
    'Q7': 'student_status',
    'Q8': 'study_level',
    'Q9': 'first_year',
    'Q10': 'field_of_study',
    'Q11': 'learning_mode',
    'Q12': 'job_confidence',

    # SECTION 2 – USAGE
    'Q13': 'used_chatgpt',
    'Q14': 'chatgpt_version',
    'Q15': 'use_frequency',
    'Q16': 'experience',
    'Q17': 'first_heard',

    # Q18 – TASK USAGE
    'Q18a': 'use_academic',
    'Q18b': 'use_professional',
    'Q18c': 'use_creative',
    'Q18d': 'use_proofreading',
    'Q18e': 'use_brainstorm',
    'Q18f': 'use_translate',
    'Q18g': 'use_summarize',
    'Q18h': 'use_calculate',
    'Q18i': 'use_study',
    'Q18j': 'use_personal',
    'Q18k': 'use_research',
    'Q18l': 'use_coding',

    # SECTION 3 – CAPABILITIES
    'Q19a': 'cap_understand',
    'Q19b': 'cap_respond',
    'Q19c': 'cap_conversation',
    'Q19d': 'cap_efficient_info',
    'Q19e': 'cap_reliable_info',
    'Q19f': 'cap_summarize',
    'Q19g': 'cap_simplify',
    'Q19h': 'cap_traditional_learning',
    'Q19i': 'cap_online_learning',
    'Q19j': 'cap_blended_learning',

    # SECTION 4 – REGULATION
    'Q20': 'institution_policy',
    'Q21a': 'reg_international',
    'Q21b': 'reg_government',
    'Q21c': 'reg_university',
    'Q21d': 'reg_employer',

    # ETHICAL CONCERNS
    'Q22a': 'ethics_unethical',
    'Q22b': 'ethics_cheating',
    'Q22c': 'ethics_plagiarism',
    'Q22d': 'ethics_threat_study',
    'Q22e': 'ethics_mislead',
    'Q22f': 'ethics_privacy',
    'Q22g': 'ethics_reduce_interaction',
    'Q22h': 'ethics_replace_education',
    'Q22i': 'ethics_isolation',
    'Q22j': 'ethics_hinder_learning',

    # ETHICAL RESPONSIBILITY
    'Q23a': 'ethics_consult_prof',
    'Q23b': 'ethics_disclose_use',
    'Q23c': 'ethics_report_misuse',
    'Q23d': 'ethics_protect_info',

    # SECTION 5 – SATISFACTION
    'Q24a': 'sat_vs_google',
    'Q24b': 'sat_vs_professors',
    'Q24c': 'sat_vs_peers',
    'Q24d': 'sat_info_clearer',
    'Q24e': 'sat_assistance',
    'Q24f': 'sat_quality',
    'Q24g': 'sat_accuracy',

    'Q25a': 'att_control',
    'Q25b': 'att_interesting',
    'Q25c': 'att_important',
    'Q25d': 'att_everyday_help',

    # SECTION 6 – ACADEMIC OUTCOMES
    'Q26a': 'acad_access_knowledge',
    'Q26b': 'acad_general_knowledge',
    'Q26c': 'acad_specific_knowledge',
    'Q26d': 'acad_personalized',
    'Q26e': 'acad_efficiency',
    'Q26f': 'acad_motivation',
    'Q26g': 'acad_complete_studies',
    'Q26h': 'acad_engagement',
    'Q26i': 'acad_deadlines',
    'Q26j': 'acad_assignment_quality',

    'Q27a': 'dev_exam_ready',
    'Q27b': 'dev_grades',
    'Q27c': 'dev_outside_class',
    'Q27d': 'dev_internships',
    'Q27e': 'dev_learning_experience',
    'Q27f': 'dev_skills',
    'Q27g': 'dev_personal_growth',
    'Q27h': 'dev_academic_growth',
    'Q27i': 'dev_study_satisfaction',
    'Q27j': 'dev_employability',

    # SECTION 7 – SKILLS
    'Q28a': 'skill_academic_writing',
    'Q28b': 'skill_prof_writing',
    'Q28c': 'skill_typing',
    'Q28d': 'skill_native_language',
    'Q28e': 'skill_foreign_language',
    'Q28f': 'skill_interpersonal',
    'Q28g': 'skill_digital_comm',
    'Q28h': 'skill_info_literacy',
    'Q28i': 'skill_content_creation',

    'Q29a': 'skill_numeracy',
    'Q29b': 'skill_decision',
    'Q29c': 'skill_problem_solving',
    'Q29d': 'skill_analytical',
    'Q29e': 'skill_critical_thinking',
    'Q29f': 'skill_creativity',
    'Q29g': 'skill_data_analysis',
    'Q29h': 'skill_programming',
    'Q29i': 'skill_ai_literacy',

    # SECTION 8 – LABOUR MARKET
    'Q30a': 'labour_reduce_jobs',
    'Q30b': 'labour_new_skills',
    'Q30c': 'labour_edu_market_link',
    'Q30d': 'labour_ai_knowledge',
    'Q30e': 'labour_ai_demand',
    'Q30f': 'labour_inequality',
    'Q30g': 'labour_create_jobs',
    'Q30h': 'labour_change_jobs',
    'Q30i': 'labour_remote_work',
    'Q30j': 'labour_reduce_workload',
    'Q30k': 'labour_productivity',
    'Q30l': 'labour_innovation',

    'Q31a': 'mismatch_skill_shortage',
    'Q31b': 'mismatch_under_skilling',
    'Q31c': 'mismatch_obsolescence',
    'Q31d': 'mismatch_skill_gap',

    # SECTION 9 – EMOTIONS
    'Q32a': 'emotion_bored',
    'Q32b': 'emotion_hopeful',
    'Q32c': 'emotion_sad',
    'Q32d': 'emotion_ashamed',
    'Q32e': 'emotion_calm',
    'Q32f': 'emotion_angry',
    'Q32g': 'emotion_relieved',
    'Q32h': 'emotion_happy',
    'Q32i': 'emotion_proud',
    'Q32j': 'emotion_anxious',
    'Q32k': 'emotion_surprised',
    'Q32l': 'emotion_curious',
    'Q32m': 'emotion_excited',
    'Q32n': 'emotion_confused',
    'Q32o': 'emotion_frustrated',

    # SECTION 10 – PERSONAL
    'Q33': 'study_difficulty',
    'Q34': 'preferred_learning_mode',

    'Q35a': 'study_success',
    'Q35b': 'study_attendance',
    'Q35c': 'study_deadlines',
    'Q35d': 'study_grade_reflect',
    'Q35e': 'study_motivation',
    'Q35f': 'financial_aid_sufficient',
    'Q35g': 'taxes_too_high',
    'Q35h': 'network_employment',
    'Q35i': 'skills_employment',
    'Q35j': 'work_experience',
    'Q35k': 'general_knowledge',
    'Q35l': 'specific_knowledge',
    'Q35m': 'faculty_employability',
    'Q35n': 'want_field_job',

    'Q36': 'area_type',
    'Q37': 'financial_aid',
    'Q38': 'employment_status',
    'Q39': 'economic_status',

    'Q40a': 'person_outgoing',
    'Q40b': 'person_helpful',
    'Q40c': 'person_neat',
    'Q40d': 'person_relaxed',
    'Q40e': 'person_curious',
    'Q40f': 'person_determined',
    'Q40g': 'person_respectful',
    'Q40h': 'person_persistent',
    'Q40i': 'person_secure',
    'Q40j': 'person_artistic',
    'Q40k': 'person_energetic',
    'Q40l': 'person_trusting',
    'Q40m': 'person_dependable',
    'Q40n': 'person_emotionally_stable',
    'Q40o': 'person_original',
}
df = df.rename(columns=rename_dict)
df.head()


citizenship  gender    age country_study  \
0  Afghanistan     2.0   27.0   Afghanistan   
1  Afghanistan     4.0  100.0   Afghanistan   
2  Afghanistan     1.0   21.0       Albania   
3  Afghanistan     1.0   18.0       Ecuador   
4  Afghanistan     2.0   26.0         Spain   

                                         institution  public_institution  \
0                                     القدس المفتوحة                 1.0   
1                                                456                 2.0   
2                                               منيا                 1.0   
3  Universidad Indoamerica., Facultad Arquitectur...                 2.0   
4                                De vigo universidad                 1.0   

   student_status  study_level  first_year  field_of_study  ...  \
0             1.0          1.0         1.0             1.0  ...   
1             2.0          3.0         2.0             4.0  ...   
2             1.0          1.0         1.0             1.0  ...   
3             1.0          1.0         1.0             4.0  ...   
4             2.0          2.0         1.0             2.0  ...   

   person_respectful  person_persistent  person_secure  person_artistic  \
0                NaN                NaN            NaN              NaN   
1                NaN                NaN            NaN              NaN   
2                3.0                2.0            3.0              4.0   
3                3.0                3.0            3.0              3.0   
4                4.0                3.0            4.0              3.0   

   person_energetic  person_trusting  person_dependable  \
0               NaN              NaN                NaN   
1               NaN              NaN                NaN   
2               3.0              2.0                2.0   
3               3.0              3.0                3.0   
4               3.0              3.0                4.0   

   person_emotionally_stable  person_original  source  
0                        NaN              NaN      AR  
1                        NaN              NaN      EN  
2                        1.0              1.0      EN  
3                        3.0              3.0      ES  
4                        3.0              4.0      ES  

[5 rows x 174 columns]

D. Find what data is missing and where.

In [107]:
total_missing = df.isna().sum().sum()
total_cells = df.shape[0] * df.shape[1]

print('Total missing values:', total_missing)
print('Percent missing overall:', round((total_missing / total_cells) * 100, 2), '%')

missing_summary = pd.DataFrame({'missing_count': df.isna().sum(),
'missing_percent': df.isna().mean() * 100
}).sort_values('missing_percent', ascending=False)

os.makedirs("data/01-interim", exist_ok=True)
df.to_csv(interim_path, index=False)

missing_summary.head()

Total missing values: 1377068
Percent missing overall: 34.09 %


missing_count  missing_percent
Q32d          10777        46.416573
Q32i          10777        46.416573
Q32g          10777        46.416573
Q32j          10774        46.403652
Q32m          10774        46.403652

From this we can see that questions later in the survey have higher missing counts and percentages.

E. Searching for outliers. For example, categories like age with any ages that are unrealistic may be of concern.

In [97]:
print(df['age'].describe())

count    23006.000000
mean        27.351343
std         91.473376
min          2.000000
25%         19.000000
50%         21.000000
75%         23.000000
max       2006.000000
Name: age, dtype: float64


F. Drop rows with missing data from vital columns. Vital columns are columns that will be neessary to know in order to analyze the data and any given relationships. Clean up the data in columns that may be incorrectly entered or too extreme to be plausible.

In [ ]:
total_missing = df.isna().sum().sum()
total_cells = df.shape[0] * df.shape[1]

print("Total missing values:", total_missing)
print("Percent missing overall:", round((total_missing / total_cells) * 100, 2), "%")

subset_vital = [
    'country_study',
    'use_frequency',
]

df = df.dropna(subset=subset_vital)

print('Shape:', df.shape)

Total missing values: 392496
Percent missing overall: 14.31 %
Shape: (15675, 175)


In [102]:
survey_year = 2023

df['age_clean'] = df['age']

# If value > 1900, treat as birth year
df.loc[df['age'] > 1900, 'age_clean'] = survey_year - df['age']

df = df[(df['age_clean'] >= 15) & (df['age_clean'] <= 100)]
df['age_clean'].describe()

count    15675.000000
mean        22.865199
std          6.268126
min         16.000000
25%         20.000000
50%         21.000000
75%         23.000000
max        100.000000
Name: age_clean, dtype: float64

G. Saving the final stage. Interim and raw stage were saved above.

In [108]:
os.makedirs("data/02-processed", exist_ok=True)
df.to_csv(processed_path, index=False)

### Dataset #2 
  - Dataset Name: Higher Education Students’ Evolving Perceptions of ChatGPT: Global Survey Data from the Academic Year 2024–2025
  - Link to the dataset: https://data.mendeley.com/datasets/nv2343nwsb/2
  - Number of observations: 22,963 students from 120 countries/territories
  - Number of variables: 175
  - The variables most relevant to our project is the self-reported frequency of students using ChatGPT for the tasks listed in Q18. The sub-questions in Q18 can be grouped into three categories: assignment production, learning support, and both. Since the responses in the dataset have already converted into numbers corresponding to the rating scale from “Never” to “Always,” the average score of each category will be utilized to compare the frequency of using ChatGPT for learning support versus assignment production.
      - Assignment production
        - academic writing (Q18a) - `use_academic`
        - professional writing (Q18b) - `use_professional`
        - creative writing (Q18c) - `use_creative`
        - proofreading (Q18d) - `use_proofreading`
      - Learning support
        - summarizing (Q18g) - `use_summarize`
        - calculating help (Q18h) - `use_calculate`
        - study assistance (Q18i) - `use_study`
        - personal assistance (Q18j) - `use_personal`
      - Both
        - brainstorming (Q18e) - `use_brainstorm`
        - translating (Q18f) - `use_translating`
        - research assistance (Q18k) - `use_reseach`
        - coding assistance (Q18l) - `use_coding`
  - Similar to dataset #1, the survey results are entirely based on the student's perception and interpretation of the questionnaire, increasing the likelihood of dishonest responses and misunderstandings. With respect to the project, the data excludes underlying specific reasons why a student would use ChatGPT to complete those tasks, such as whether one uses it for the convenience, or due to the competitiveness of their academic environment, which potentially ties to the economic development of a country. Furthermore, this dataset lacks reports on the students’ financial circumstances. Therefore, we also need to incorporate datasets that display the economic development of the countries reported by the students in order to get a clearer visualization that can answer our questions.

In [ ]:
## YOUR CODE TO LOAD/CLEAN/TIDY/WRANGLE THE DATA GOES HERE


## Ethics

Instructions: Keep the contents of this cell. For each item on the checklist
-  put an X there if you've considered the item
-  IF THE ITEM IS RELEVANT place a short paragraph after the checklist item discussing the issue.
  
Items on this checklist are meant to provoke discussion among good-faith actors who take their ethical responsibilities seriously. Your teams will document these discussions and decisions for posterity using this section.  You don't have to solve these problems, you just have to acknowledge any potential harm no matter how unlikely.

Here is a [list of real world examples](https://deon.drivendata.org/examples/) for each item in the checklist that can refer to.

[![Deon badge](https://img.shields.io/badge/ethics%20checklist-deon-brightgreen.svg?style=popout-square)](http://deon.drivendata.org/)

### A. Data Collection
 - [X] **A.1 Informed consent**: If there are human subjects, have they given informed consent, where subjects affirmatively opt-in and have a clear understanding of the data uses to which they consent?

People who took the survey took it knowingly. One concern is that they might not have known their data would be used in our project.

 - [X] **A.2 Collection bias**: Have we considered sources of bias that could be introduced during data collection and survey design and taken steps to mitigate those?

 Survey takers took it willingly so there may be response bias in the answers.

 - [X] **A.3 Limit PII exposure**: Have we considered ways to minimize exposure of personally identifiable information (PII) for example through anonymization or not collecting information that isn't relevant for analysis?

 Anonymizing and identifiable information

 - [X] **A.4 Downstream bias mitigation**: Have we considered ways to enable testing downstream results for biased outcomes (e.g., collecting data on protected group status like race or gender)?

 By using data from multiple countries and territories, we are able to get a diverse range of data and survey takers. We also take into consideration their own perceptions rather than others perceptions of them.

### B. Data Storage
 - [X] **B.1 Data security**: Do we have a plan to protect and secure data (e.g., encryption at rest and in transit, access controls on internal users and third parties, access logs, and up-to-date software)?

 We will anonymize any sensitive data that may directly or indirectly reveal the identity of any individuals, including device identifiers, names, addresses, etc. If the dataset is leaked, those who took the survey may be at risk of exposing confidential information to unauthorized organizations and individuals, thus at risk for identity theft and doxxing and such.

 - [X] **B.2 Right to be forgotten**: Do we have a mechanism through which an individual can request their personal information be removed?

 The individual can contact us through email or other stated contact to request the removal of their personal information. Their personal information may be required to prove themselves as the owner of the data they wish to remove.

 - [X] **B.3 Data retention plan**: Is there a schedule or plan to delete the data after it is no longer needed?

 After the data has been used for its original purpose, it should be deleted unless it has been explicitly stated at the beginning of data collection and the individual is notified explicitly. This helps prevent the risk of data breaches and leak of personal data.

### C. Analysis
 - [X] **C.1 Missing perspectives**: Have we sought to address blindspots in the analysis through engagement with relevant stakeholders (e.g., checking assumptions and discussing implications with affected communities and subject matter experts)?
 - [X] **C.2 Dataset bias**: Have we examined the data for possible sources of bias and taken steps to mitigate or address these biases (e.g., stereotype perpetuation, confirmation bias, imbalanced classes, or omitted confounding variables)?
 - [X] **C.3 Honest representation**: Are our visualizations, summary statistics, and reports designed to honestly represent the underlying data?
 - [X] **C.4 Privacy in analysis**: Have we ensured that data with PII are not used or displayed unless necessary for the analysis?
 - [X] **C.5 Auditability**: Is the process of generating the analysis well documented and reproducible if we discover issues in the future?

When analyzing the data, it’s important to consider both direct and indirect factors that could affect how students use ChatGPT. While a country’s level of economic development is the main focus, other variables like field of study, language background, and access to technology could influence ChatGPT usage at the same time. If these factors are ignored, the results could be misleading. Including them helps reduce confounding and makes the findings more applicable across different student groups. 

The dataset is also checked for potential bias. Since the data are self-reported, the same students may not report their ChatGPT usage accurately, either because they are unsure how to categorize their use or because they are worried about academic consequences. Some countries may also have fewer responses than others, which could skew the comparisons. These issues are addressed by checking for imbalance groups, avoiding strong conclusions from small samples, and being transparent about uncertainty in the results. Visualizations and summary statistics are kept simple and clearly labeled so the data are not exaggerated or misrepresented. 

Student privacy is protected throughout the analysis by not collecting or displaying any personally identifiable information. All results are reported at an aggregate level, and no individual student responses are singled out. Finally, the analysis process is documented step by step, including how variables are created and how decisions are made, so the work can be reviewed or repeated later if problems are found. This helps keep the analysis transparent and accountable.


### D. Modeling
 - [X] **D.1 Proxy discrimination**: Have we ensured that the model does not rely on variables or proxies for variables that are unfairly discriminatory?
 - [X] **D.2 Fairness across groups**: Have we tested model results for fairness with respect to different affected groups (e.g., tested for disparate error rates)?
 - [X] **D.3 Metric selection**: Have we considered the effects of optimizing for our defined metrics and considered additional metrics?
 - [X] **D.4 Explainability**: Can we explain in understandable terms a decision the model made in cases where a justification is needed?
 - [X] **D.5 Communicate limitations**: Have we communicated the shortcomings, limitations, and biases of the model to relevant stakeholders in ways that can be generally understood?

Because this project studies differences in how students across countries use ChatGPT, it is important to make sure that the model does not rely on variables that unfairly discriminate against certain groups or act as proxies for sensitive characteristics. Country-level economic indicators are necessary for the analysis, but they are only used to describe broader context, not to make judgments about individual students. The model avoids using variables that could indirectly stand in for sensitive characteristics, and we check relationships between variables to make sure none of them function as unfair proxies. To make sure the model behaves reasonably across different groups, results are compared for students from different income-level countries, academic fields, and language backgrounds to see whether error patterns differ in meaningful ways. Instead of relying on a single performance metric, we look at multiple measures so that overall accuracy does not hide uneven performance across groups. 

We also prioritize models that are easy to interpret, which makes it possible to explain in simple terms how economic development relates to both the frequency and type of ChatGPT use. Throughout the analysis, we clearly acknowledge limitations such as self-reported data, possible sampling bias, and the fact that the results show associations rather than casual relationships.


### E. Deployment
 - [X] **E.1 Monitoring and evaluation**: Do we have a clear plan to monitor the model and its impacts after it is deployed (e.g., performance monitoring, regular audit of sample predictions, human review of high-stakes decisions, reviewing downstream impacts of errors or low-confidence decisions, testing for concept drift)?
 - [X] **E.2 Redress**: Have we discussed with our organization a plan for response if users are harmed by the results (e.g., how does the data science team evaluate these cases and update analysis and models to prevent future harm)?
 - [X] **E.3 Roll back**: Is there a way to turn off or roll back the model in production if necessary?
 - [X] **E.4 Unintended use**: Have we taken steps to identify and prevent unintended uses and abuse of the model and do we have a plan to monitor these once the model is deployed?

Even though this project is mainly academic, the findings could still be misused if they are taken out of context. For example, country-level trends might be used to make assumptions about individual students or to justify unequal treatment, which is not what the data support. To reduce this risk, the results should be reviewed periodically, especially if ChatGPT usage patterns change over time or if new data are added. If harm or misuse is identified, the analysis should be revisited and, if necessary, revised or withdrawn. Since the model is not meant for automated or high-stakes decisions, rolling it back would simply involve stopping its use or removing it from any downstream applications. Being clear about appropriate and inappropriate uses of the results, and monitoring how they are shared or interpreted, is essential to preventing unintended harm.


## Team Expectations 

- **Communication:** Reply within 24 hours; if anything urgent come up the last minute, please notify in the group chat asap
- **Discussion:** Try to meet at least once over the weekend and/or on the Tuesday over zoom before the next checkpoint is due
- **Decision:** made based on a majority vote; but exception may apply
- **Work distribution:** If any members is assigned a less challenging part, they should be flexible to assist other members on their assigned parts when work is completed
- **Editing:** If two or more members are working on the code on the same part, notify in the group chat to receive confirmation before committing and pushing

## Project Timeline

| Meeting Date  | Meeting Time| Completed Before Meeting  | Discuss at Meeting |
|---|---|---|---|
| 2/1 |  iMessage | Read & Think about COGS 108 expectations; brainstorm topics/questions  | (iMessage gc) Determine best form of communication; Discuss and decide on final project topic; discuss hypothesis; begin background research | 
| 2/3 |  9 PM on Zoom |  Do background research on topic; Edit, finalize, and submit proposal| Discuss ideal dataset(s) and ethics; draft project proposal |
| 2/10 |  9 PM on Zoom | Revise proposal for resubmission | Confirm datasets; Discuss Wrangling and possible analytical approaches; re-Assign group members to lead each specific part |
| Wk 6 Thu-Wknd |  iMessage | Complete dataset description | Discuss Wrangling and possible analytical approaches; Check-in for proposal revision |
| 2/18 | 9 PM on Zoom  |  Complete (or almost completed) DataCheckpoint  |  Check-in before push; discuss EDACheckpint  |
| 2/25 | TBD | Revision of previous submission; Import & Wrangle Data (Ant Man); EDA (Hulk) | Review/Edit wrangling/EDA; Discuss Analysis Plan  |
| 3/3 | TBD | Finalize wrangling/EDA; Begin Analysis (Iron Man; Thor) | Discuss/edit Analysis |
| Weekend before Wk 10 | TBD | Complete analysis; Draft results/conclusion/discussion (Wasp)| Discuss/edit full project |
| Weekend before Finals Wk | TBD  | Finish filming; getting started on editing| Turn in Final Project & Group Project Surveys |